src/bokeh/models/glyphs.py

```diff
@@ -859,6 +860,27 @@ def __init__(self, *args, **kwargs) -> None:
         The color mapping step happens on the client.
     """)
 
+    def _validate_with_data_source(self, source: ColumnDataSource):
+        class_name = self.__class__.__name__
+
+        if not isinstance(self.color_mapper, WeightedStackColorMapper):
+            raise ValueError(f"{class_name}.color_mapper should be a WeightedStackColorMapper not a {self.color_mapper.__class__.__name__}")
```

Comment:
This should have been handled in the type of the property. Also the description of `color_mapper` property is wrong, probably copied from `ImageRGBA`.